docs/web/usecases-internationalization.mdx

```diff
@@ -0,0 +1,280 @@
+
+```ts title="src/index.ts"
+import { ComponentType } from "react";
+import { LibraryRegistry } from "@geocortex/web/config";
+import TranslatableText, {
+    TranslatableTextModel,
+} from "./components/TranslatableText";
+// highlight-start
+import enJson from "./en.json";
+import deJson from "./de.json";
+// highlight-end
+
+export default function (registry: LibraryRegistry) {
+    registry.registerComponent({
+        name: "translatable-text",
+        namespace: "your.custom.namespace",
+        getComponentType: () => TranslatableText as ComponentType,
+        itemType: "translatable-text",
+        title: "Translatable Text",
+    });
+    registry.registerModel({
+        getModelType: () => TranslatableTextModel,
+        itemType: "translatable-text",
+    });
+    // highlight-start
+    registry.registerLanguageResources({
+        locale: "en",
+        getValues: () => enJson,
+    });
+    registry.registerLanguageResources({
+        locale: "de",
+        getValues: () => deJson,
+    });
+    // highlight-end
+}
+```
+
+### Use the Language Strings
+
+Now that they have been registered, we can use the strings in our boilerplate component from earlier. To translate a language key, you can use the [`useContext` react hook](https://reactjs.org/docs/hooks-reference.html#usecontext) with the [UIContext](sdk-ui-context.mdx) to access the `translate` function. Then you can pass the language key `language-translatable-text-content` you defined in the language resources files to the `translate` function and your text will be translated into whichever locale is most appropriate.
+
+<!-- TODO-API link to Geocortex UI Components -->
+
+Another option for translating language strings is to use Geocortex UI Components, like `<Typography>`. Geocortex Components will automatically detect and translate language strings in props and child content.
+
+:::tip
+To test out a translation, add the URL parameter `locale=<language>` to your local server.
+
+For example, `http://localhost:3000/?locale=de`
+:::
+
+<Tabs
+  defaultValue="component"
+  values={[
+    { label: "Component", value: "component" },
+    { label: "English UI", value: "english" },
+    { label: "German UI", value: "german" },
+  ]}>
+<TabItem value="component">
+
+```tsx title="src/components/TranslatableText/TranslatableText.tsx"
+import React, { useContext } from "react";
+import { LayoutElement } from "@geocortex/web/components";
+// highlight-next-line
+import { UIContext } from "@geocortex/web/ui";
+import Typography from "@geocortex/web/ui/typography";
+
+export default function TranslatableText(props) {
+    // highlight-next-line
+    const { translate } = useContext(UIContext);
+
+    return (
+        <LayoutElement {...props} style={{ backgroundColor: "white" }}>
+            // highlight-start
+            <h1>{translate("language-translatable-text-content")}</h1>
```

Comment:
Given this sample code, the screenshots should be showing the title twice, shouldn't it? It struck me as odd that they don't.

Reply:
Good catch. Updated the screenshots